bot.py

```diff
@@ -1,274 +1,91 @@
+import traceback
 
-        with open(join(day_path, file), "r") as f:
-            input = f.read()
-
-        status = await msg.reply(f"Benchmarking input {i+1}")
-        out = await run_image(msg, input)
-        if not out:
-            return
-        await status.delete()
-
-        result = {}
-        for line in out.splitlines():
-            if line.startswith("FERRIS_ELF_ANSWER "):
-                result["answer"] = str(line[18:]).strip()
-            if line.startswith("FERRIS_ELF_MEDIAN "):
-                result["median"] = int(line[18:])
-            if line.startswith("FERRIS_ELF_AVERAGE "):
-                result["average"] = int(line[19:])
-            if line.startswith("FERRIS_ELF_MAX "):
-                result["max"] = int(line[15:])
-            if line.startswith("FERRIS_ELF_MIN "):
-                result["min"] = int(line[15:])
-
-        if verify:
-            if not result["answer"] == verify:
-                await msg.reply(f"Error: Benchmark returned wrong answer for input {i + 1}")
-                return
-            verified = True
-        else:
-            print("Cannot verify run", result["answer"])
-
-        cur.execute("INSERT INTO runs VALUES (?, ?, ?, ?, ?, ?, ?)", (str(msg.author.id), code, day, part, result["median"], result["answer"], result["answer"]))
-        results.append(result)
-    
-
-    median = avg([r["median"] for r in results])
-    average = avg([r["average"] for r in results])
+import discord
 
-    if verified:
-        await msg.reply(embed=discord.Embed(title="Benchmark complete", description=f"Median: **{ns(median)}**\nAverage: **{ns(average)}**"))
-    else:
-        await msg.reply(embed=discord.Embed(title="Benchmark complete (Unverified)", description=f"Median: **{ns(median)}**\nAverage: **{ns(average)}**"))
+import lib
+from messages import SubmitMessage, GetBestTimesMessage
+from database import Database
 
-    db.commit()
-    print("Inserted results into DB")
+from config import settings
+import constants
 
 class MyBot(discord.Client):
     queue = asyncio.Queue()
+    db = Database().get()
 
     async def on_ready(self):
         print("Logged in as", self.user)
-
         while True:
             try:
-                msg = await self.queue.get()
-                print(f"Processing request for {msg.author.name}")
-                code = await msg.attachments[0].read()
-                parts = [p for p in msg.content.split(" ") if p]
-                day = int((parts[0:1] or (today(), ))[0])
-                part = int((parts[1:2] or (1, ))[0])
-
-                await benchmark(msg, code, day, part)
-
+                submit_msg = await self.queue.get()
+                print(f"Going to process submission from queue: {submit_msg}")
+                await lib.benchmark(submit_msg)
                 self.queue.task_done()
             except Exception as err:
-                print("Queue loop exception!", err)
+                traceback.print_exception(err)
```

Comment:
Probably an existing bug, but can we verify this exception handling doesn't create an infinite loop? Docs links or experimental evidence are both good avenues.

Reply:
I have tested this. The task is removed from the queue when calling `get()`, so it's not going to be around even if processing failed. The `task_done` is more of a notification mechanism for inter thread communication.